### PR TITLE
Add some domain related traits

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

File: include/ddc/traits.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "non_uniform_discretization.hpp"
+#include "uniform_discretization.hpp"
+
+template <class T>
+struct is_uniform_domain : std::false_type
+{
+};
+
+template <class... DDims>
+struct is_uniform_domain<DiscreteDomain<DDims...>>
```
Comment (jbigot) [inline include/ddc/traits.hpp:14]: shouldn't you include something for `DiscreteDomain`?

Comment (tpadioleau) [reply include/ddc/traits.hpp:14]: I think a declaration is missing in this case.

Comment (jbigot) [reply include/ddc/traits.hpp:14]: Not sure to understand

Comment (tpadioleau) [reply include/ddc/traits.hpp:14]: I mean this:
```suggestion

template <class... DDims>
class DiscreteDomain<DDims...>;

struct is_uniform_domain<DiscreteDomain<DDims...>>
```

**Review by jbigot — APPROVED** [review]

**Pull request merged**